cmd/downloader/downloader/downloader.go

```diff
@@ -325,12 +371,17 @@ func MainLoop(ctx context.Context, d *Downloader, silent bool) {
 				t.AllowDataDownload()
 				t.DownloadAll()
 				go func(t *torrent.Torrent) {
+					defer sem.Release(1)
 					//r := t.NewReader()
 					//r.SetReadahead(t.Length())
 					//_, _ = io.Copy(io.Discard, r) // enable streaming - it will prioritize sequential download
 
 					<-t.Complete.On()
-					sem.Release(1)
+					go func(t *torrent.Torrent) {
```

Comment:
Why do you need nested `go func` here? Perhaps you mean to do one of them without `go` to use `defer sem.Release()` ?

Reply:
You right. I don’t need. It’s trash after experiment.